tests/extmod_hardware/machine_uart_irq_rxidle.py

```diff
@@ -0,0 +1,71 @@
+# Test machine.UART.IRQ_RXIDLE firing after a set of characters are received.
+#
+# IMPORTANT: This test requires hardware connections: the UART TX and RX
+# pins must be wired together.
+
+try:
+    from machine import UART
+
+    UART.IRQ_RXIDLE
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+import time, sys
+
+# Configure pins based on the target.
+if "esp32" in sys.platform:
+    uart_id = 1
+    tx_pin = 4
+    rx_pin = 5
+elif "mimxrt" in sys.platform and "Teensy 4" in sys.implementation._machine:
+    uart_id = 1
+    tx_pin = None
+elif "pyboard" in sys.platform:
+    uart_id = 4
+    tx_pin = None  # PA0
+    rx_pin = None  # PA1
+elif "renesas-ra" in sys.platform:
+    uart_id = 9
+    tx_pin = None  # P602 @ RA6M2
+    rx_pin = None  # P601 @ RA6M2
+elif "rp2" in sys.platform:
+    uart_id = 0
+    tx_pin = "GPIO0"
+    rx_pin = "GPIO1"
+elif "samd" in sys.platform and "ItsyBitsy M0" in sys.implementation._machine:
+    uart_id = 0
+    tx_pin = "D1"
+    rx_pin = "D0"
+    byte_by_byte = True
+elif "samd" in sys.platform and "ItsyBitsy M4" in sys.implementation._machine:
+    uart_id = 3
+    tx_pin = "D1"
+    rx_pin = "D0"
+else:
+    print("SKIP")
+    raise SystemExit
+
+
+def irq(u):
+    if u.irq().flags() & u.IRQ_RXIDLE:
```

Comment:
Is stm32 the only port that needs this check?  If so then I suggest to remove the check and let the stm32 port fail.  That will serve as a reminder to fix stm32 (in a separate PR).

Reply:
Removed.